<?xml version="1.0"?>
<launch>

	<arg name = "Driver_Port" default = "/dev/ttyUSB0" />
	<arg name = "Driver_Baud" default = "115200" />
	<!-- <arg name = "bluetooth" default = "/dev/Bluetooth" /> -->
	<arg name = "control_hz" default = "20" />
	<arg name = "Hokuyo_IP"   default = "192.168.44.10" />
	
	<node pkg="willy_v2" type="willy_v2_base" name="willy_base" respawn="true" output="screen">
		<param name="Baudrate" value="$(arg Driver_Baud)" />
		<param name="Port" value="$(arg Driver_Port)" />
        <param name="control_hz" value="$(arg control_hz)" />
	</node>

	<node pkg="willy_v2" type="willy_v2_encoder" name="willy_encoder" respawn="true" >
		<param name="Baudrate" value="$(arg Driver_Baud)" />
		<param name="Port" value="$(arg Driver_Port)" />
		<!-- <remap from = "/odom" to = "/odom_data"/> -->
	</node>

        <!--node pkg="joy_control" type="joy_function.py" name="joy_function" output="screen"/-->

	<!-- <include file = "$(find joy_control)/launch/joy_control.launch" />
	<include file = "$(find urg_node)/launch/urg_lidar.launch" /> -->


	<!-- <node pkg="willy_v2" type="willy_v2_fault_detection" name="willy_v2_fault_detection"/> -->
		
    	<!--node pkg="willy" type="main.py" name="willy_UI"/-->

	<!--node  pkg="urg_node" type="urg_node" name="urg_node" >
		<param name="ip_addressalue" value="$(arg Hokuyo_IP)" />
		<param name="angle_min" value="-2"/>
    		<param name="angle_max" value="2"/>
	</node-->
    
        
	<!--include file = "$(find rplidar_ros)/launch/rplidar.launch" /-->		

	<!--node pkg="rqt_plot" name="plot_right_vel" type="rqt_plot" args=" /right_vel /left_vel"/               output="screen"     -->

</launch>
